src/db/service/ExecutionService.ts

```diff
@@ -1,11 +1,142 @@
+      });
+    });
+    new Procedures().triggerProcedure(procedure, parameters);
+    test.rules.forEach(async (rule: RuleModelType) => {
+      rule.unitTests.forEach(async (unitTest: UnitTestType) => {
+        const { expectedRecordMatches, total, expectedNumRecords, table } =
+          unitTest;
+        switch (unitTest.level) {
+          case UnitTestOperations.TableGenericOperations: {
+            const res = await new TableTestService().check(
+              unitTest as TableTestType
+            );
+            if (unitTest.operation === TableGenericOperations.EXISTS) {
+              unitTest.result = Boolean(res);
+              unitTest.output = `${unitTest.table} table exists is ${unitTest.result}`;
+            } else {
+              switch (expectedRecordMatches) {
+                case RecordMatches.ZERO:
+                  unitTest.result = Number(res) === 0;
+                  unitTest.output = `${unitTest.table} had ${Number(
+                    res
+                  )} rows and expected 0 rows`;
+                  break;
+                case RecordMatches.GREATER_THAN:
+                  unitTest.result = Number(res) > 0;
+                  unitTest.output = `${unitTest.table} had ${Number(
+                    res
+                  )} rows and expected more than 0 rows`;
+                  break;
+                case RecordMatches.TABLE_ROWS:
+                  if (total) {
+                    unitTest.result = Number(res) === expectedNumRecords;
+                    unitTest.output = `${unitTest.table} had ${Number(
+                      res
+                    )} rows and expected ${expectedNumRecords} rows`;
+                  } else {
+                    unitTest.result =
+                      Number(res) - tableRows[table] === expectedNumRecords;
+                    unitTest.output = `${unitTest.table} had ${
+                      Number(res) - tableRows[table]
+                    } new rows and expected ${expectedNumRecords} newrows`;
+                  }
+                  break;
+                default:
+                  unitTest.result = false;
+                  break;
+              }
+            }
+            break;
+          }
+          case UnitTestOperations.RowBooleanOperations: {
+            const rows = await new RowTestService().check(
+              unitTest as RowTestType
+            );
+            switch (expectedRecordMatches) {
+              case RecordMatches.ZERO:
+                unitTest.result = Number(rows?.length) === 0;
+                unitTest.output = JSON.stringify(rows);
+                unitTest.output = `First 10 rows: ${JSON.stringify(
+                  rows?.slice(0, 10)
+                )}`;
+                break;
+              case RecordMatches.GREATER_THAN:
+                unitTest.result = Number(rows?.length) > 0;
+                unitTest.output = `First 10 rows: ${JSON.stringify(
+                  rows?.slice(0, 10)
+                )}`;
```

Comment:
If you just leave this as the JSON string then in front end they can display it easily. Also there's a `format` field that can be either `string` or `json` that you should set so they know what to format it as